src/InputField/index.js

```diff
@@ -323,29 +322,88 @@ const InputField = React.forwardRef((props: Props, ref: Ref) => {
     inputMode,
   } = props;
 
-  const forID = id || (label ? React.useMemo(() => randomID("inputFieldID"), []) : undefined);
+  const forID = React.useMemo(() => {
+    if (id) return id;
+    return label ? randomID("inputFieldID") : undefined;
+  }, [id, label]);
+  const [tooltipShown, setTooltipShown] = useState(false);
+  const [tooltipShownHover, setTooltipShownHover] = useState(false);
+  const labelRef = useRef(null);
+  const iconRef = useRef(null);
 
   return (
     <Field
       component={label ? "label" : "div"}
       spaceAfter={spaceAfter}
       htmlFor={label ? forID : undefined}
     >
-      {label && !inlineLabel && <FormLabel label={label} isFilled={!!value} required={required} />}
-      <InputContainer size={size} disabled={disabled} error={error}>
-        {prefix && <Prefix size={size}>{prefix}</Prefix>}
+      {label && !inlineLabel && (
+        <FormLabel
+          inlineLabel={inlineLabel}
+          filled={!!value}
+          required={required}
+          error={!!error}
+          help={!!help}
+          labelRef={labelRef}
+          iconRef={iconRef}
+          onMouseEnter={() => setTooltipShownHover(true)}
+          onMouseLeave={() => setTooltipShownHover(false)}
+        >
+          {label}
+        </FormLabel>
+      )}
+      <InputContainer
+        size={size}
+        disabled={disabled}
+        error={error}
+        labelRef={label ? null : labelRef}
+      >
+        {prefix &&
```

Comment:
It seems quite complex, don't you wanna create seperate sub-component for it?

Reply:
I would just move the complexity somewhere else, not sure is needed.

Reply:
I have updated the condition slightly, due to new changes. I think it's more readable now, let me know if further simplification is needed :))